pkg/kubelet/kubelet.go

```diff
@@ -2124,6 +2178,43 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 	}
 }
 
+func (kl *Kubelet) checkPodResourcesResize(pod *v1.Pod) {
+	podResized := false
+	for _, container := range pod.Spec.Containers {
+		if len(diff.ObjectDiff(container.ResourcesAllocated, container.Resources.Requests)) > 0 {
```

Comment:
should the comparison with container.Resouces.limits instead of the requested values ?

Reply:
not here. this is at the entry point of handling pod update. if user changes the desired resources.requests and resources.limits, first check is to verify that new requests (resource reservation on the node) still fits the node available resources. if it does, then node patches the resourcesAllocated in apiserver pod-spec confirming that node  accepts the desired change. limits is checked in SyncPod, we always drive container towards desired limits (check that limit change is valid, does not change QoS etc happens in validation)